### ISSUE TITLE
Virtual layer with 'WHERE ... IN (...)' clause optimization

### ISSUE THREAD

**Issue by uclaros** [qgis/QGIS#35520]
**Feature description.**
A virtual layer defined with a `WHERE "field" IN ('value_1', ..., 'value_n')` clause will spawn n feature requests with clauses `WHERE "field" = 'value_n'`.
This can pass unnoticed on local connections but on remote and/or slow connections this can cause significant delays.
A single feature request should be passed to the underlying provider.

**Additional context**
Tested with a virtual layer querying a single oracle table.

**Comment by gioman** [comment]
Good catch. As for https://github.com/qgis/QGIS/issues/35381 this to me is a borderline bug. This type of things could indeed explain a lot of the pain users feel when using remote connections.
